src/client/shader.cpp

```diff
@@ -140,8 +141,25 @@ class SourceShaderCache
 			type = "geometry";
 		else if (str_ends_with(filename, ".fs.glsl"))
 			type = "fragment";
-		else
+		else {
 			errorstream << "Unsupported shader type: " << filename << std::endl;
+			return;
+		}
+		if (program.length() > 10000) {
+			errorstream << "Rejecting long shader (> 10k chars): " << filename << std::endl;
+			return;
+		}
+
+		const std::regex line_comments("//.*?\n");
```

Comment:
`static const`, and maybe `optimize` (somewhere in regex_constants, don’t remember).

Reply:
Done.